keydates-worker/test_keydates_worker.py

```
@@ -1071,5 +1072,359 @@ def test_closes_still_moves_earlier(self):
+
+    def test_retry_after_nonfinite_or_negative_falls_back_and_retries(self):
+        # F1: "inf" overflows int(float(...)) (OverflowError) and "-5" would reach
+        # time.sleep(-5); both must fall back to the 60s retry, not crash chat().
+        body = {"choices": [{"message": {"content": json.dumps({"dates": []})}}]}
+        for header in ("inf", "1e400", "-5"):
+            with self.subTest(retry_after=header):
+                calls = {"n": 0}
+                slept = []
+
+                def fake(req, timeout=None):
+                    calls["n"] += 1
```

Ruff (0.16.1)

[warning] Function definition does not bind loop variable `calls`

(B023)